### DISCUSSION TITLE
How to update curl version to 8.4.0 in rke2 kube-system images

### DISCUSSION THREAD

**Discussion by SamaDevOps** [rancher/rke2#5930] [Q&A] [answered]
Hi!!
We have an airgapped rke2 cluster with version v1.28.5+rke2r1. In the rke2 kube-system images, the cURL version is 8.0.1. Under CVE-2023-38545, it is suggested to upgrade cURL to 8.4.0. But even in the newer images, I can only see the cURL version of 8.0.1 as the latest version. Is CVE-2023-38545 not affecting rke2? If there is no images with cURL 8.4.0 how do we manually update the cURL in the kube-system images?

**Comment by brandond** [comment, marked as answer]
We're currently in the process of replacing the base image for RKE2 system images with a more minimal image. Kubernetes doesn't need or use curl, or many of the other things, that are currently present in the base image.

A smaller base image will allow for smaller RKE2 system images, and a reduced attack surface / vulnerability profile. Less things installed means less things to update.

**Reply by SamaDevOps** [reply]
Hi @brandond 
Thank you for the quick reply! could I know the timeline of the base image replacement or a discussion related to that?

**Reply by manuelbuil** [reply]
Most of the images have already been replaced by smaller ones in ` v1.28.9+rke2r1`. Could you give it a try?

**Reply by SamaDevOps** [reply]
Hi @manuelbuil We will look into the v1.28.9+rke2r1 version. Thank you for the reply!

**Reply by manuelbuil** [reply]
I noticed cloud-provider was not updated using the scratch base image. We will get the update for the June release: https://github.com/rancher/rke2/pull/5933
